### PR TITLE
Add geometryID to NGE

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14171/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14171/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14171/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14171/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14171/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14171/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14171/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14171/merge#BCU1XR#0

File: packages/dev/core/src/Meshes/mesh.vertexData.ts [hunk 1]
```diff
@@ -114,6 +114,8 @@ export class VertexData {
      */
     public static readonly DEFAULTSIDE = 0;
 
+    private static _UniqueIDGenerator = 0;
```
Comment (RaananW) [inline packages/dev/core/src/Meshes/mesh.vertexData.ts:117]: If it didn't have to be consecutive, you could simply use the scene unique if mechanism.

**Review by RaananW — APPROVED** [review]
One small comment, otherwise LGTM

File: packages/dev/core/src/Meshes/mesh.vertexData.ts (hunk 1 above)
Comment (RaananW) [reply packages/dev/core/src/Meshes/mesh.vertexData.ts:117]: Ignore that, the scene is not really available when it's constructed

Comment (deltakosh) [reply packages/dev/core/src/Meshes/mesh.vertexData.ts:117]: yep :) less dependencies ;D

**Pull request merged**